### PR TITLE
Add `asset-assembler` pkg in monorepo setup

### PR DESCRIPTION
**Pull request by jkrems**
A very basic implementation of batching chunks that I have been using when prototyping with webpack/parcel/next/etc.. What's maybe more interesting than the actual code for assembly is the overall setup that is somewhat modeled after the way angular/webpack/next/parcel (the primary "partner repos") are set up. In particular:

1. Code is authored in TypeScript.
2. Individual packages are in `packages/`. Dependencies managed using yarn workspaces.
3. Use of the native `composite`/`references` feature for TypeScript to get incremental builds across packages.
4. Tests using jest, mirroring the TypeScript setup with projects.
5. Lerna for publishing new versions, auth handled via the wombat dressing room.

Roughly: "yarn workspace" is a "tsc (project) reference" is a "jest project". For sharing project settings there is `packages/jest.base.js` and `packages/tsconfig.base.json`.

### CODE REVIEW

**Comment by jkrems** [comment]
@soldair Would love to hear your thoughts - am I missing something important? I assume "lint and/or prettier" is the obvious answer..?

File: packages/asset-assembler/src/assemble.spec.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import path from 'path';
```
Comment (fenghaolw) [inline packages/asset-assembler/src/assemble.spec.ts]: Should we have some top-level comments about this file?

File: packages/asset-assembler/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
+'use strict';
```
Comment (fenghaolw) [inline packages/asset-assembler/src/index.ts]: similarly should we have comments for each export: some of them are not obvious from the variable name.

File: packages/asset-assembler/src/assemble.spec.ts (hunk 1 above)
Comment (jkrems) [reply packages/asset-assembler/src/assemble.spec.ts]: Added a summary to this file and also a README to the example chunksets to explain their structure.

File: packages/asset-assembler/src/index.ts (hunk 2 above)
Comment (jkrems) [reply packages/asset-assembler/src/index.ts]: Done! Added comments to all exported types/functions.

**Pull request merged**